python/docs/source/getting_started/install.rst

```diff
@@ -165,16 +168,117 @@ To install PySpark from source, refer to |building_spark|_.
+
+
+.. _optional-dependencies:
+
+Optional dependencies
+~~~~~~~~~~~~~~~~~~~~~
+
+PySpark has several optional dependencies that enhance its functionality for specific modules.
+These dependencies are only required for certain features and are not necessary for the basic functionality of PySpark.
+If these optional dependencies are not installed, PySpark will function correctly for basic operations but will raise an ``ImportError``
+when you try to use features that require these dependencies.
+
+Spark SQL
+^^^^^^^^^
+
+Installable with ``pip install "pyspark[sql]"``.
+
+========= ================= ==================================================================
+Package   Supported version Note
+========= ================= ==================================================================
+`pandas`  >=1.4.4           Enables seamless DataFrame operations between Spark and Pandas.
+`pyarrow` >=10.0.0          Optimizes data conversion and transfer between PySpark and Pandas.
+`numpy`   >=1.21            Essential for numerical data manipulation within PySpark.
+========= ================= ==================================================================
+
+
+Pandas API on Spark
+^^^^^^^^^^^^^^^^^^^
+
+Installable with ``pip install "pyspark[pandas_on_spark]"``.
+
+========= ================= =====================================================================
+Package   Supported version Note
+========= ================= =====================================================================
+`pandas`  >=1.4.4           Required for utilizing the Pandas API features in Spark.
+`pyarrow` >=10.0.0          Ensures efficient data handling and performance in Pandas operations.
+`numpy`   >=1.21            Facilitates complex numerical operations within Spark.
+========= ================= =====================================================================
+
+Additional libraries that enhance functionality but are not included in the installation packages:
+
+- **mlflow**: Enhances machine learning lifecycle management, including experiment tracking and model deployment.
+- **plotly, matplotlib**: Provide advanced plotting capabilities for visualization.
+
+
+ML
+^^
+
+Installable with ``pip install "pyspark[ml]"``.
+
+======= ================= =======================================================================
+Package Supported version Note
+======= ================= =======================================================================
+`numpy` >=1.21            Supports advanced data manipulation and algorithm implementation in ML.
+======= ================= =======================================================================
+
+Additional libraries that enhance functionality but are not included in the installation packages:
+
+- **scipy**: Essential for scientific computing and statistical functions in ML.
+- **scikit-learn**: Required for implementing machine learning algorithms.
```

Comment:
@WeichenXu123 and @zhengruifeng can you help writing better description?

Reply:
I think we can use this description first.